### PR TITLE
Adding name and namespace headers.

### PR DESCRIPTION
**Pull request by GhostofCookie**

### CODE REVIEW

File: include/quicr/quicr_namespace.h [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#pragma once
```
Comment (GhostofCookie) [inline include/quicr/quicr_namespace.h:1]: Missing operators

File: include/quicr/quicr_name.h [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    std::string to_hex() const;
+
+    Name operator>>(uint64_t value);
+    Name operator<<(uint64_t value);
+    Name operator+(uint64_t value);
+    Name operator-(uint64_t value);
+    Name operator&(uint64_t value);
+    Name operator|(uint64_t value);
+    Name operator&(const Name& other);
+    Name operator|(const Name& other);
```
Comment (GhostofCookie) [inline include/quicr/quicr_name.h:31]: Maybe useless given the uint64 version?

File: include/quicr/quicr_namespace.h [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    bool contains(const Namespace& name_space) const;
+
+    friend bool operator==(const Namespace& a, const Namespace& b);
+    friend bool operator!=(const Namespace& a, const Namespace& b);
+    friend bool operator>(const Namespace& a, const Namespace& b);
+    friend bool operator<(const Namespace& a, const Namespace& b);
+
+private:
+    Name _name;
+    uint64_t _sig_bits{120};
```
Comment (fluffy) [inline include/quicr/quicr_namespace.h]: don't default this and I think given the valid range is 0 to 127 , probably should just be uint16_t

File: include/quicr/quicr_name.h [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    Name(const std::vector<uint8_t>& data);
+    Name(const Name& other);
+    Name(Name&& other);
+    ~Name() = default;
+
+    std::vector<uint8_t> data() const;
+    size_t size() const;
+    std::string to_hex() const;
+
+    Name operator>>(uint64_t value);
```
Comment (fluffy) [inline include/quicr/quicr_name.h]: for the shit operatioins, seems like value to shift by could be uint16_t

**Review by fluffy — APPROVED** [review]
Looks great.

**Pull request merged**